hack/install-wrapper.sh

```diff
@@ -0,0 +1,250 @@
+    --ns)
+    NAMESPACE="$2"
+    shift 2
+    ;;
+    --kubeconfig)
+    KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-kubeconfig)
+    ANTREA_KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-version)
+    ANTREA_VERSION="$2"
+    shift 2
+    ;;
+    --bin)
+    AGENT_BIN="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    print_help
+    exit 1
+    ;;
+esac
+done
+
+if [ -z "$ANTREA_VERSION" ] || [ -z "$NAMESPACE" ] || [ -z "$KUBECONFIG" ] || [ -z "$ANTREA_KUBECONFIG" ]; then
+  echo "Required fields are not set"
+  print_usage
+  exit 1
+fi
+
+# Generic function to execute command with timeout.
+function execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date)"
+        echoerr "Exiting!! please check internet connectivity and re-run install script"
+        exit 2
+    fi
+}
+
+function get_cloud_platform() {
+    cat /sys/class/dmi/id/bios_version | grep -iq "amazon"
+    if [ $? -eq 0 ]; then
+        echo -n "$AWS"
+        return
+    fi
+    cat /sys/class/dmi/id/bios_vendor | grep -iq "amazon"
+    if [ $? -eq 0 ]; then
+        echo -n "$AWS"
+        return
+    fi
+    cat /sys/class/dmi/id/sys_vendor | grep -iq "microsoft"
+    if [ $? -eq 0 ]; then
+        echo -n "$AZURE"
+        return
+    fi
+}
+
+function check_os_prerequisites() {
+    CLOUD=$(get_cloud_platform)
+    if [ -z "$CLOUD" ]; then
+        echoerr "Unknown cloud platform, supported cloud platforms: AWS and Azure"
+        exit 2
+    fi
+    OS=$(grep -Po '^ID=\K.*' /etc/os-release | sed -e 's/^"//' -e 's/"$//')
+    echo "Installing antrea-agent on $OS, cloud $CLOUD"
+    case "$OS" in
+        ubuntu)
+          return
+        ;;
+    esac
+    echoerr "Unsupported $OS platform"
+    exit 2
+}
+
+function install_curl() {
+    echo "Installing curl in $OS"
+    case "$OS" in
+        ubuntu)
+            cmd="apt-get install curl -y"
+            execute_command "$cmd" 1m
+        ;;
+    esac
+}
+
+function install_required_packages() {
+    if ! command -v curl &> /dev/null; then
+        install_curl
+    fi
+}
+
+function update_antrea_url() {
+    ANTREA_BRANCH="release-${ANTREA_VERSION}"
+    ANTREA_TAG="v${ANTREA_VERSION}.0"
+    ANTREA_INSTALL_SCRIPT="https://raw.githubusercontent.com/antrea-io/antrea/${ANTREA_BRANCH}/hack/externalnode/install-vm.sh"
+    ANTREA_CONFIG="https://raw.githubusercontent.com/antrea-io/antrea/${ANTREA_BRANCH}/build/yamls/externalnode/conf/antrea-agent.conf"
+    if [ -z "$AGENT_BIN" ]; then
+        AGENT_BIN="https://github.com/antrea-io/antrea/releases/download/${ANTREA_TAG}/antrea-agent-x86_64.exe"
+    fi
+}
+
+function download_file() {
+    from=$1
+    to=$2
+    echo "Downloading file $from to $to"
```

Comment:
Why have execute command function if you arent using it for generic purpose?

Reply:
Decided not to use it for now.